File: erpsynch_app/lib/sync/hasher.dart
import 'dart:convert';
import 'dart:io';
import 'package:crypto/crypto.dart';

String hashFile(String filePath) {
  try {
    final bytes = File(filePath).readAsBytesSync();
    return sha256.convert(bytes).toString();
  } catch (_) {
    return '';
  }
}

String hashContent(List<int> bytes) {
  return sha256.convert(bytes).toString();
}

String hashString(String content) {
  return sha256.convert(utf8.encode(content)).toString();
}

File: src/hasher.ts
import fs from 'fs';
import crypto from 'crypto';

export function hashFile(filePath: string): string {
  try {
    const content = fs.readFileSync(filePath);
    return crypto.createHash('sha256').update(content).digest('hex');
  } catch {
    return '';
  }
}

export function hashContent(content: string | Buffer): string {
  return crypto.createHash('sha256').update(content).digest('hex');
}
